### ISSUE TITLE
Change in Carbon parse functionality

### ISSUE THREAD

**Issue by arborrow** [arborrow/montserrat#448]
It appears there was a change in how Carbon::parse handles the YYYYMMDD string. It expects a time with dashes YYYY-MM-DD. So first I want to understand what changed in Carbon with version 2.41.0 to make sure that it is not a regression. Secondly, I want to review how I am both creating buttons with dates in the route to make sure that any use of Carbon::parse is successful. At the moment, in the PageController for the cash and credit card reports, I am checking for an 8-digit string and adding the dashes but this is rather hacky so I should just generate the date properly so that they contain the dashes in route path.

**Comment by arborrow** [comment]
Looks like https://github.com/briannesbitt/Carbon/issues/2214 is at the root of the problem and I suspect it will be fixed. Still not a bad idea to clean up in Polanco so that we can avoid this unexpected behavior. It seems to easy to confuse the YYYYMMDD format with a UNIX timestamp value. I prefer to avoid the ambiguity. We be nice to do a try and call an exception (404) if the parse fails rather than a 500.

**Comment by arborrow** [comment]
This issue has been resolved upstream - see https://github.com/briannesbitt/Carbon/issues/2214#issuecomment-706532313 - although it will be good to review the web routes and make sure that we are using dashes throughout Polanco as that seems to be the best practice.

**Comment by arborrow** [comment]
A review of the web routes file indicates that the following routes may make use of dates:
Route::get('person/{contact_id}/eoy_acknowledgment/{start_date?}/{end_date?}', 'PageController@eoy_acknowledgment');
Route::get('finance/cash_deposit/{day?}', 'PageController@finance_cash_deposit')->name('report.finance.cash_deposit');
Route::get('finance/cc_deposit/{day?}', 'PageController@finance_cc_deposit')->name('report.finance.cc_deposit');
Route::get('rooms/{ym?}/{building?}', 'RoomController@schedule')->name('rooms');

**Comment by arborrow** [comment]
I've searched the Controllers for any use of Carbon's parse function and want to minimize the use of it. When possible I eliminated using it for store and update methods by ensuring the date field is cast in the model (for example touched_at in Touchpoint). In cases where I was allowing the route to specify an 8 digit date (yyyymmdd) I created a hyphenate_date function in PageController to add the hyphens (yyyy-mm-dd). A similar approach is used in the Room controller.
